runner/entrypoint.sh

```diff
@@ -24,6 +24,11 @@ if [ ! -z "${STARTUP_DELAY_IN_SECONDS}" ]; then
   sleep ${STARTUP_DELAY_IN_SECONDS}
 fi
 
+if [ -n "${WAIT_FOR_DOCKER}" ]; then
+  echo "Waiting for docker to be available"
+  until docker info; do sleep 1; done
+  echo "Docker is now available"
+fi
```

Comment:
We would like this to be mroe automatic so the end user doesn't need to set this setting, it should be automatic if the runner has docker enabled. 

If you add to the below to [here](https://github.com/actions-runner-controller/actions-runner-controller/blob/master/controllers/runner_controller.go#L777) you can check if docker is enabled in the entrypoint. If it is then we should wait for docker to be avaliable, if it isn't then we shouldn't.

add this:

```go
		{
			Name:  "DOCKER_ENABLED",
			Value: fmt.Sprintf("%v", dockerEnabled),
		},
```

before this:

```go
		{
			Name:  "DOCKERD_IN_RUNNER",
			Value: fmt.Sprintf("%v", dockerdInRunner),
		},

```

Reply:
Do we want an option to opt out? Or just an always on option.

Why does a new controller need to be built and tested?

Reply:
I'll see if I have time for that change this week. I was able to get this quick change with some free time last week, but I have less of that.

Reply:
> Do we want an option to opt out? Or just an always on option.

To be honest I'm not why you would want to opt out of preventing a race condition. @mumoshu if you have any thoughts.

Reply:
I think the default `WAIT_FOR_DOCKER` value should be dependent on `DOCKER_ENABLED`(=dockerEnabled from the controller) and I do also think an ability to opt-out `WAIT_FOR_DOCKER` would be a nice-to-have.

We don't yet have a controller functionality to auto-detect the version of GHES, so opting-out WAIT_FOR_DOCKER would be the only way to keep actions-runner-controller working on old versions of GHES after this change.

Reply:
As for the opt-out, we could use "0" as the opt-out, and any other int we could interpret as yes we should wait for docker with the provided value beeing the upper limit (timeout) in seconds before we terminate from the startup script.

We could really use this as well, currently running with a fixed startup delay to account for docker (but just this morning some jobs failed because docker wasn't up in time). This would be a great addition/improvment